frontend_tests/node_tests/list_render.js

```diff
@@ -731,3 +734,98 @@ run_test('opts.get_item', () => {
         ['two', 'three']
     );
 });
+
+run_test('render item', () => {
+    const container = make_container();
+    const scroll_container =  make_scroll_container(container);
+    const INITIAL_RENDER_COUNT = 80; // Keep this in sync with the actual code.
+    container.html = () => {};
+    let called = false;
+    scroll_container.find = (query) => {
+        const expected_queries = [`tr[data-item='${INITIAL_RENDER_COUNT + 1}']`, "tr[data-item='3']"];
+        const new_html = '<tr data-item=3>updated: 3</tr>\n';
+        const regex = /\<tr data-item=3\>.*?\<\/tr\>/;
+        assert(expected_queries.includes(query));
+        if (query.includes(`data-item='${INITIAL_RENDER_COUNT + 1}'`)) {
+            return; // This item is not rendered, so we find nothing
+        }
+        return {
+            // Return a JQuery stub for the original HTML.
+            // We want this to be called when we replace
+            // the existing HTML with newly rendered HTML.
+            replaceWith: (html) => {
+                assert.equal(new_html, html);
+                called = true;
+                container.appended_data.replace(regex, new_html);
+            },
+        };
+    };
+
+    const list = [...Array(100).keys()];
+
+    let text = 'initial';
+    const get_item = (item) => ({text: `${text}: ${item}`, value: item});
+
+    const widget = list_render.create(container, list, {
+        name: 'replace-list',
+        modifier: (item) => `<tr data-item=${item.value}>${item.text}</tr>\n`,
+        get_item: get_item,
+        html_selector: (item) => `tr[data-item='${item}']`,
+    });
+
+    assert(container.appended_data.html().includes('<tr data-item=2>initial: 2</tr>'));
+    assert(container.appended_data.html().includes('<tr data-item=3>initial: 3</tr>'));
+    text = 'updated';
+    called = false;
+    widget.render_item(3);
+    assert(called);
+    assert(container.appended_data.html().includes('<tr data-item=2>initial: 2</tr>'));
+    assert(container.appended_data.html().includes('<tr data-item=3>updated: 3</tr>'));
+
+    // Item 81 should not be in the rendered list.
```

Comment:
let's test item-80 case directly next to the item-81 code, to avoid item-81 becoming a test that doesn't really prove anything (in the future)